public/app/plugins/datasource/tempo/traceql/QueryEditor.tsx

```diff
@@ -1,25 +1,42 @@
+import { css } from '@emotion/css';
 import { defaults } from 'lodash';
-import React from 'react';
+import React, { useState } from 'react';
 
 import { QueryEditorProps } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
 
 import { TempoDatasource, TempoQuery } from '../datasource';
 import { defaultQuery, MyDataSourceOptions } from '../types';
 
+import { TempoQueryBuilderOptions } from './TempoQueryBuilderOptions';
 import { TraceQLEditor } from './TraceQLEditor';
 
 type Props = QueryEditorProps<TempoDatasource, TempoQuery, MyDataSourceOptions>;
 
 export function QueryEditor(props: Props) {
-  function onEditorChange(value: string) {
-    props.onChange({ ...props.query, query: value });
-  }
+  const [query, setQuery] = useState(defaults(props.query, defaultQuery));
```

Comment:
We don't need to create a new state for the query since that's already being managed by the TempoDatasource. The `query` object and the `onChange` function are already passed as `props` and that's all you'll need to use for this component. Check [PromQueryEditorSelector](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.tsx#L24) and also [PromQueryBuilderOptions](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilderOptions.tsx#L29) to see the best way to achieve this functionality.

Reply:
I have refactored to use the `onChange` function in commit https://github.com/grafana/grafana/pull/54435/commits/79c99554d7b079a1e221e4d03fd0a929553ed8aa.